src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -288,13 +288,16 @@ const useAppPolling = (googleProject, workspaceName) => {
   }
   const loadApps = async () => {
     try {
-      const newApps = !!googleProject ?
+      const newGoogleApps = !!googleProject ?
         await Ajax(signal).Apps.list(googleProject, { role: 'creator', saturnWorkspaceName: workspaceName }) :
         []
-      setApps(newApps)
+      const newAzureApps = !!workspaceId ? await Ajax(signal).Apps.getV2AppInfo(workspaceId) : []
```

Comment:
Shouldn't you only call this endpoint for Azure workspaces? Note that `isGoogleWorkspace(workspace)` and `isAzureWorkspace(workspace)` is preferred over checking `googleProject`.

Reply:
I think this will work as expected, because a google project will return an empty list of V2 apps... but I agree that a workspace type check could quickly short-circuit without an additional API call if it is a google workspace

Reply:
It seems like this really should be a conditional API call based on which type of workspace it is, and then you don't need to combine apps together.

Reply:
That's a good point. I will make them conditional calls based on whether its a google project or not.

Reply:
I tried this out and it seems that when this method is defined the `workspace` variable hasn't been defined yet and this causes the `isGoogleWorkspace(workspace)` to fail with `Cannot read properties of undefined (reading 'workspace')` error. It seems that was the reason to use `!!googleProject`. I can switch the code to `isGoogleWorkspace` and `isAzureWorkspace` like this and it works:
```
const newGoogleApps = !!workspace && isGoogleWorkspace(workspace) ? await Ajax(signal).Apps.list(googleProject, { role: 'creator', saturnWorkspaceName: workspaceName }) : []
const newAzureApps = !!workspace && isAzureWorkspace(workspace) ? await Ajax(signal).Apps.getV2AppInfo(workspace.workspace.workspaceId) : []
```